pkgs/development/beam-modules/fetch-mix-deps.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenvNoCC, elixir, hex, rebar, rebar3, cacert }:
+
+{ name, version, sha256, src, mixEnv ? "prod", debug ? false, meta ? { } }:
+
+with stdenvNoCC.lib;
+
+stdenvNoCC.mkDerivation ({
+  name = "mix-deps-${name}-${version}";
+
+  phases = [ "configurePhase" "downloadPhase" ];
+
+  nativeBuildInputs = [ elixir hex cacert ];
+
+  inherit src;
+
+  MIX_ENV = mixEnv;
+  MIX_REBAR = "${rebar}/bin/rebar";
+  MIX_REBAR3 = "${rebar3}/bin/rebar3";
+  MIX_DEBUG = if debug then 1 else 0;
+
+  configurePhase = ''
+    mkdir -p $out/deps
+    mkdir -p $out/.hex
+    export HEX_HOME="$out/.hex";
+    export MIX_HOME="$TEMPDIR/.mix";
+    export MIX_DEPS_PATH="$out/deps";
+  '';
+
+  downloadPhase = ''
+    ln -s ${src}/mix.exs ./mix.exs
+    ln -s ${src}/mix.lock ./mix.lock
+    mix deps.get --only ${mixEnv}
+  '';
+
+  outputHashAlgo = "sha256";
+  outputHashMode = "recursive";
+  outputHash = sha256;
+
+  impureEnvVars = stdenvNoCC.lib.fetchers.proxyImpureEnvVars;
+  inherit meta;
+})
```

Comment:
I sometime see fetchers adding `} // removeAttrs args [ "pname" "version", ... ])` at the end so that the fetcher is more "hackable", should that apply here? Do we need to run some postConfigure/Download hooks?

Reply:
Interested to hear other people's opinion. On my end I can't see use case for making it more hackable. Interesting idea though.

Reply:
I could see this potentially being useful for packages with C or other non-BEAM sources.

Reply:
I like this feedback. I would say, for the time being, just in the interest of getting this over the line, let's proceed with the current course and we can change this when it is needed.